### PR TITLE
Proper category errors from serverApi and publicApi IDE-162

### PR DESCRIPTION
**Pull request by mustard-mh**
## Related Issue(s)

Fixes [IDE-162](https://linear.app/gitpod/issue/IDE-162)

## How to test

✅ https://github.com/gitpod-io/gitpod-vscode-desktop/pull/80#issuecomment-1592464464

Add break points and test with cases below

- Server API with random workspaceID
- Server API without auth token
- Public API with random workspaceID
- Public API without auth token

## Documentation

/hold

### CODE REVIEW

**Comment by mustard-mh** [comment]
- Server API with random workspaceID
  image
- Server API without auth token
 image2
- Public API with random workspaceID
  image
- Public API without auth token
  image

File: src/internalApi.ts [hunk 1]
```diff
@@ -84,9 +91,46 @@ class GitpodServerApi extends vscode.Disposable {
...
+			return Code.ResourceExhausted;
+		case 470:
+			return Code.PermissionDenied;
+		case -32603:
+			return Code.Internal;
+	}
+	if (err.code >= 400 && err.code < 500) {
+		return Code.InvalidArgument;
+	}
+	return;
```
Comment (akosyakov) [inline src/internalApi.ts:135]: what if err.code >= 500 shold not we return internal?

**Review by akosyakov — APPROVED** [review]
let's try

File: src/internalApi.ts (hunk 1 above)
Comment (mustard-mh) [reply src/internalApi.ts:135]: https://github.com/gitpod-io/gitpod/blob/d41a38ba83939856e5292e30912f52e749787db1/components/public-api-server/pkg/proxy/errors.go#LL25C1-L26C1

👍 Will do it

File: src/publicApi.ts [hunk 2]
```diff
@@ -66,11 +67,24 @@ export class GitpodPublicApi extends Disposable implements IGitpodAPI {
...
+                }
+                throw err;
+            }
+        };
 
         const transport = createConnectTransport({
             baseUrl: serviceUrl.toString(),
             httpVersion: '2',
-            interceptors: [authInterceptor, metricsInterceptor],
+            interceptors: [errorWrapInterceptor, authInterceptor, metricsInterceptor],
```
Comment (mustard-mh) [inline src/publicApi.ts:87]: `errorWrap` should be loaded before `metrics` in case they break each other

**Pull request merged**